core/templates/dev/head/player/PlayerServices.js

```diff
@@ -535,69 +542,76 @@ oppia.factory('oppiaPlayerService', [
           answer, currentInteraction.id, currentInteraction.customization_args);
       }
     },
-    submitAnswer: function(answer, successCallback, delayParamUpdates) {
+    getClassificationResult: function(answer) {
       if (answerIsBeingProcessed) {
         return;
       }
 
       _numSubmitsForThisState++;
 
-      answerIsBeingProcessed = true;
       var oldState = angular.copy(_exploration.states[_currentStateName]);
 
-      answerClassificationService.getMatchingClassificationResult(
+      return answerClassificationService.getMatchingClassificationResult(
         _explorationId, oldState, answer
-      ).success(function(classificationResult) {
-        var outcome = classificationResult.outcome;
-
-        if (!_editorPreviewMode) {
-          var answerRecordingUrl = (
-            '/explorehandler/answer_submitted_event/' + _explorationId);
-          $http.post(answerRecordingUrl, {
-            answer: answer,
-            params: learnerParamsService.getAllParams(),
-            version: version,
-            old_state_name: _currentStateName,
-            rule_spec_string: classificationResult.rule_spec_string
-          });
-        }
+      );
+    },
+    processClassificationResult: function(
```

Comment:
I feel like this function is doing too much. In fact, to me it seems it's doing the following three things:

1. Recording a submitted answer.
1. Handling nth-resubmission triggers.
1. Preparing to transition to the next state.

These should probably be three separate functions.